src/server/redis_reply.cc

```diff
@@ -37,9 +37,23 @@ namespace redis {
 
 void Reply(evbuffer *output, const std::string &data) { evbuffer_add(output, data.c_str(), data.length()); }
 
-std::string SimpleString(const std::string &data) { return "+" + data + CRLF; }
+std::string SimpleString(std::string_view data) {
+  std::string res;
+  res.reserve(data.size() + 3);  // 1 for '+', 2 for CRLF
+  res += RESP_PREFIX_SIMPLE_STRING;
+  res += data;
+  res += CRLF;
+  return res;
+}
 
-std::string Error(const Status &s) { return RESP_PREFIX_ERROR + StatusToRedisErrorMsg(s) + CRLF; }
+std::string Error(const Status &s) {
+  std::string res;
+  res.reserve(s.Msg().size() + 3);  // 1 for '-', 2 for CRLF
```

Comment:
I wonder that `s.Msg()` might not be really the final string generation when `redisErrorPrefixMapping` in `StatusToRedisErrorMsg` contains the specific error message.

Reply:
Yes, you're absolutely right; I overlooked the possibility that the text could come directly from redisErrorPrefixMapping. However, performing a hash lookup here to confirm the reserved length might be a bit heavy.

Or, do you think it’s a good idea to directly use std::max(16, s.Msg().size() + 3)? In most cases, the minimum allocation unit by the memory allocator is often two pointer lengths, and for the contents within redisErrorPrefixMapping, 16 bytes should be sufficient.

Reply:
> However, performing a hash lookup here to confirm the reserved length might be a bit heavy.

As the hash table is small and anyway it would be built 😅, I think construct error string firstly is ok...

> do you think it’s a good idea to directly use std::max(16, s.Msg().size() + 3)

It would be good if it's a critical path, but I don't think the error path worth this, since the current code is clear to understand and Error path might not a bottleneck...

Reply:
Ok, I've reverted this modification.